feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/search/SearchResults.kt

```diff
@@ -0,0 +1,154 @@
+import androidx.compose.foundation.layout.only
+import androidx.compose.foundation.layout.safeDrawing
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ListItem
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Snackbar
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.odyssey.model.SearchableClaim
+import com.hedvig.android.odyssey.R
+
+@Composable
+fun ClaimsSearchResults(
+  viewState: SearchViewState,
+  onClaimSelected: (SearchableClaim) -> Unit,
+  onClaimNotCovered: (String) -> Unit,
+  cantFindAddress: () -> Unit,
+  closeKeyboard: () -> Unit,
+) {
+  Column {
+    SuggestionsList(
+      viewState = viewState,
+      onClaimSelected = { claim ->
+        closeKeyboard()
+        onClaimSelected(claim)
+      },
+      onClaimNotCovered = onClaimNotCovered,
+      cantFindAddress = cantFindAddress,
+      contentPadding = WindowInsets.safeDrawing
+        .only(WindowInsetsSides.Horizontal.plus(WindowInsetsSides.Bottom))
+        .asPaddingValues(),
+      modifier = Modifier.weight(1f),
+    )
+  }
+}
+
+@OptIn(ExperimentalMaterialApi::class, ExperimentalFoundationApi::class)
+@Composable
+private fun SuggestionsList(
+  viewState: SearchViewState,
+  onClaimSelected: (SearchableClaim) -> Unit,
+  onClaimNotCovered: (String) -> Unit,
+  cantFindAddress: () -> Unit,
+  contentPadding: PaddingValues,
+  modifier: Modifier = Modifier,
+) {
+  LazyColumn(
+    modifier = modifier,
+    contentPadding = contentPadding,
+  ) {
+    items(
+      items = viewState.results,
+      key = { item: SearchableClaim -> item.id },
+      contentType = { "ItemEntry" },
+    ) { claim: SearchableClaim ->
+      ListItem(
+        icon = {
+          Image(
+            painter = painterResource(claim.icon()),
+            contentDescription = "Covered",
+            modifier = Modifier
+              .size(24.dp),
+          )
+        },
+        secondaryText = {
+          if (claim.hasQuickPayout) {
+            Text(text = "Quick payout")
+          } else if (!claim.isCovered) {
+            Text(text = "Not covered")
+          } else {
+            Text(text = "Start a claim")
+          }
+        },
+        text = { Text(text = claim.displayName) },
+        modifier = Modifier
+          .animateItemPlacement()
+          .clickable {
+            if (claim.isCovered) {
+              onClaimSelected(claim)
```

Comment:
I wonder for these, if it would make it more complicated or less complicated if this composable just had something like `claimClick : (SearchableClaim) -> Unit` and the decision on what is happening is determined on a higher up point, maybe even the activity, so that the compose code can focus as much as possible on just the UI and not the business logic.
Not 100% sure though, this isn't something that I've myself settled on and every time find myself thinking about it in different ways.
Feel free to do nothing or tell me your thoughts on this if you have a stronger opinion on it.